src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp

```diff
@@ -128,10 +129,20 @@ struct YarpRobotControl::Impl
             return false;
         }
 
-        if (!this->controlModeInterface->getControlModes(this->controlModesYarp.data()))
+        // try to read the control mode
+        unsigned counter = 0;
+        constexpr unsigned maxIter = 100;
+        constexpr unsigned timeout = 500;
```

Comment:
Is it worth putting them as part of the initialization parameters?

Reply:
We can have default values and the user can change them. Do you agree?

Reply:
Absolutely

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/118/commits/78738d2c5d8b12dd4d27669a07074a17e02073de